pulpcore/cli/ansible/repository.py

```diff
@@ -32,18 +33,15 @@
 _ = gettext.gettext
 
 
-def _remote_callback(
-    ctx: click.Context, param: click.Parameter, value: Optional[str]
-) -> Optional[str]:
-    # Pass None and "" verbatim
-    if value:
-        pulp_ctx = ctx.find_object(PulpContext)
-        assert pulp_ctx is not None
-        try:
-            return PulpAnsibleCollectionRemoteContext(pulp_ctx, entity={"name": value}).pulp_href
-        except click.ClickException:
-            return PulpAnsibleRoleRemoteContext(pulp_ctx, entity={"name": value}).pulp_href
-    return value
+remote_option = resource_option(
+    "--remote",
+    default_plugin="ansible",
+    default_type="collection",
+    context_table={
+        "ansible:collection": PulpAnsibleCollectionRemoteContext,
+        "ansible:role": PulpAnsibleRoleRemoteContext,
+    },
+)
```

Comment:
@gerrod3 
This needs an update of the ansible sync workflow:

https://github.com/pulp/pulp_ansible/blob/master/docs/_scripts/sync.sh#L2